front/pages/w/[wId]/builder/data-sources/[name]/index.tsx

```diff
@@ -638,6 +642,82 @@ function SlackBotEnableView({
   );
 }
 
+function GithubCodeEnableView({
+  owner,
+  readOnly,
+  isAdmin,
+  dataSource,
+}: {
+  owner: WorkspaceType;
+  readOnly: boolean;
+  isAdmin: boolean;
+  dataSource: DataSourceType;
+}) {
+  const { configValue, mutateConfig } = useConnectorConfig({
+    owner,
+    dataSource,
+    configKey: "codeSyncEnabled",
+  });
+  const codeSyncEnabled = configValue === "true";
+
+  const sendNotification = useContext(SendNotificationsContext);
+  const [loading, setLoading] = useState(false);
+
+  const handleSetCodeSyncEnabled = async (codeSyncEnabled: boolean) => {
+    setLoading(true);
+    const res = await fetch(
+      `/api/w/${owner.sId}/data_sources/${dataSource.name}/managed/config/codeSyncEnabled`,
+      {
+        headers: {
+          "Content-Type": "application/json",
+        },
+        method: "POST",
+        body: JSON.stringify({ configValue: codeSyncEnabled.toString() }),
+      }
+    );
+    if (res.ok) {
+      await mutateConfig();
+      setLoading(false);
+    } else {
+      setLoading(false);
+      const err = (await res.json()) as { error: APIError };
+      sendNotification({
+        type: "error",
+        title: "Failed to enable GitHub code sync",
+        description: err.error.message,
+      });
+    }
+    return true;
+  };
+
+  return (
+    <ContextItem.List>
+      <ContextItem
+        title="Code Synchronization"
+        visual={<ContextItem.Visual visual={GithubLogo} />}
+        action={
+          <div className="relative">
+            <SliderToggle
+              size="xs"
+              onClick={async () => {
+                await handleSetCodeSyncEnabled(!codeSyncEnabled);
+              }}
+              selected={codeSyncEnabled || false}
```

Comment:
Why the default to false?

Reply:
cargo culted might not be needed, checking.

Reply:
(removed)